static/js/narrow.js

```diff
@@ -265,6 +265,49 @@ exports.activate = function (raw_operators, opts) {
     }, 0);
 };
 
+exports.by_stream_topic_name = function (stream, topic) {
+    var stream_id = stream_data.get_stream_id(stream);
+
+    if (stream_id === undefined) {
+        return;
+    }
+
+    var topics = topic_data.get_recent_names(stream_id);
+
+    var opts = {
+        select_first_unread: true,
+    };
+
+    var raw_operators = [
+        {operator: 'stream', operand: stream},
+    ];
+
+    if (topic !== '' && _.indexOf(topics, topic) >= 0) {
+        raw_operators.push({operator: 'topic', operand: topic});
+    }
+
+    exports.activate(raw_operators, opts);
+};
+
+exports.by_recipient_name = function (recipient) {
+    var private_recipients = util.extract_pm_recipients(recipient);
+    var invalid_recipients = _.reject(private_recipients, people.is_valid_email_for_compose);
+
+    var opts = {
+        select_first_unread: true,
+    };
+
+    if (invalid_recipients.length !== 0 || recipient === '') {
```

Comment:
It's usually good style to do early-return for the "error" condition, not the "normal" condition.  And then it's not clear to me what the rest of this function is supposed to do.

Reply:
I will add some comments here.